File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.js
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
const reversePrefix = (word, ch) => {
  let i = word.indexOf(ch);
  if (~i) {
    ++i;
    return word.substring(0, i).split('').reverse().join('') + word.substring(i);
  }
  return word;
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
# class Solution:
#   def reversePrefix(self, word: str, ch: str) -> str:
#     i = word.find(ch)
#     return word if i < 0 else word[i::-1] + word[i + 1:]

class Solution:
  def reversePrefix(self, word: str, k: str) -> str:
    res = ''
    found = False
    for ch in word:
      if found:
        res += ch
      else:
        res = ch + res
        if ch == k:
          found = True
    return res if found else word
        